idseq_dag/steps/generate_phylo_tree.py

```diff
@@ -21,11 +21,24 @@ class PipelineStepGeneratePhyloTree(PipelineStep):
     and/or
       (b) Genbank full reference genomes from the same taxid.
     '''
+
+    k_config = {
+        # Chosen kmer-size as a function of kingdom.
+        # Must be an odd number.
+        # All entries to be revisited and benchmarked.
+        # Values for viruses and bacteria come from kSNP3 recommendations (13-15 / 19-21).
+        "Viruses": 13,
+        "Bacteria": 19,
+        "Eukaryota": 19,
+        None: 13
+    }
+
     def run(self):
         output_files = self.output_files_local()
         taxid = self.additional_attributes["taxid"]
         reference_taxids = self.additional_attributes.get("reference_taxids", [taxid]) # Note: will only produce a result if species-level or below
         superkingdom_name = self.additional_attributes.get("superkingdom_name")
+        self.k = PipelineStepGeneratePhyloTree.k_config[superkingdom_name]
```

Comment:
maybe self.k_superkingdom or something?

Reply:
Hmm I don't think so. Right now we choose k simply based on superkingdom, but in general that could change in the future

Reply:
got it